haskellPackages: update stackage and hackage

sternenseemann opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).
  
We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.
  
### haskellPackages Workflow Summary
  
Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).
  
The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer ha
s been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

---
    
This is the follow-up to #219027. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

sternenseemann committed:
haskellPackages: stackage LTS 20.12 -> LTS 20.14

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

sternenseemann committed:
all-cabal-hashes: 2023-03-01T16:43:25Z -> 2023-03-13T08:59:39Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

sternenseemann committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

Kranzes committed:
cachix: remove no longer needed src override & patch

The patch and overrides removed in this change were only needed
temporarly to avoid waiting for a haskell-updates cycle.

Kranzes committed:
cachix,hercules-ci-{agent,cnix-expr,cnix-store}: bump nix pin to 2_14

Kranzes committed:
haskellPackages.tomland: unbreak

Kranzes committed:
Merge pull request #221036 from Kranzes/haskell-updates

{cachix,hercules-ci}: remove no longer needed override and update passthru.nixPackage

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.Cabal_3_10_1_0: build with matching of Cabal-syntax

sternenseemann committed:
haskellPackages.Cabal{,-syntax}_3_8_1_0: keep

Latest version is 3.10.1.0 now, but let's keep 3.8 around (for now at
least). We will likely soon have moved a lot of stuff over to 3.10
though.

sternenseemann commented:
Getting cabal-install 3.10.1.0 to work for GHC < 9.4 looks to be quite annoying at the moment (but probably possible).

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskell.packages.ghc92.th-desugar: use stackage prescribed version

sternenseemann committed:
haskell.packages.ghc94.th-desugar: 1.14 -> 1.15

Also add override for GHC 9.6, 1.15 is the first version to support GHC
9.6. The override likely doesn't work yet, but is added in preparation.

To make it compile, we need a matching version of th-abstraction.

sternenseemann committed:
haskell.packages.ghc94.ghc-lib*: retain GHC 9.4 versions

These were dropped by hackage2nix since the GHC 9.6 versions were
uploaded to Hackage.

sternenseemann committed:
haskellPackages.haskell-gi: drop obsolete override

Stackage LTS has updated to 0.26.3 as well!

sternenseemann committed:
haskellPackages.hspec*: reflect 2.10.9 -> 2.10.10

Commit:
haskellPackages.haddock-cheatsheet: unbreak

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
Merge pull request #221748 from KAction/contrib/0/haddock-cheatsheet/out

haskellPackages.haddock-cheatsheet: unbreak

sternenseemann committed:
haskell.packages.ghc94.indexed-traversable*: drop obsolete overrides

sternenseemann committed:
haskell.packages.ghc96.cabal-install: make work

Forces us to get basic 9.6 support up (which was quite painful). Even
more pain is waiting for us when trying to get cabal-install to build
with GHCs < 9.6…

sternenseemann committed:
haskell.packages.ghc94.generically: supply base-orphans

base-orphans is required for building generically with
generically >= 9.4 && < 9.6.

sternenseemann committed:
release-haskell.nix: test weeder with GHC 9.4

sternenseemann committed:
haskell.packages.ghc94.aeson: drop obsolete jailbreak

sternenseemann committed:
haskell.packages.ghc96.ghc-lib*: pick appropriate versions

sternenseemann committed:
release-haskell.nix: test GHC 9.6 package set

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskell.packages.ghc94.ghc: 9.4.2 -> 9.4.4

Due to a mistake in 9e8a48377099 the default 9.4 GHC versions wasn't
properly adjusted for the package set.

sternenseemann committed:
haskell.packages.ghc94.cabal-install: make work

This simply needs the override for Cabal, Cabal-syntax 3.10. All other
core packages match the ones expected by cabal-install and friends.

sternenseemann committed:
release-haskell.nix: test Cabal-syntax >= 3.8 as well

domenkozar committed:
Merge branch 'master' into haskell-updates

domenkozar committed:
fix cachix build

sternenseemann committed:
haskell.packages.ghc94.weeder: don't test on aarch64

cborg does not compile with GHC >= 9.4 and aarch64

sternenseemann committed:
haskell.packages.*.vector: clean up and unify overrides, run tests

- The dontCheck overrides are not necessary completely. It seems that
  there is some kind of issue with the doctest suite, but we can run the
  other ones.

- We need a jailbreak everywhere due to doctest (ironically), so we can
  move that into the common configuration.

- The i686 issue has been closed upstream. Tested locally to confirm
  that it is indeed not a problem anymore.

sternenseemann committed:
haskell.packages.ghc94.cborg: resolve duplicate overlay entry

sternenseemann committed:
haskell.packages.ghc94.cborg: unrestrict platforms

Revert "haskell.packages.ghc94.cborg: resolve duplicate overlay entry"

This reverts commit 6634553e48f47185056d6646784a5f8fd329b1fe.

Revert "haskell.packages.ghc94.weeder: don't test on aarch64"

This reverts commit 600fd969f6d317524f4e859824f577663d795d15.

Reason for reverts: https://github.com/NixOS/nixpkgs/pull/221688 applied
a patch for the compilation failure we were seeing on aarch64.

sternenseemann committed:
haskell.packages.ghc92.cabal-install: get to work

The directory >= 1.3.7.0 constraint proved to be a real troublemaker.
Since directory is a core package other core packages frequently depend
upon (as opposed to process) which tends to cause inconsistent
dependency errors.

To get it to work the following was done:

- Use directory 1.3.7.1 to eliminate the need to also upgrade filepath
  and unix, saving a bit of trouble. This still forces us to recompile
  process every time directory is upgraded (i.e. GHC >= 9.2.5) even if
  the core package process version would match to prevent inconsistent
  dependencies.

- Upgrade hspec to 2.10 where it no longer depends on ghc and
  ghc-boot-th which naturally causes inconsistent dependencies by
  depending on core packages, namely directory.

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
haskellPackages.hermes-json_0_2_0_1: generate package

cdepillabout committed:
nix-output-monitor: use hermes-json-0.2.0.1

sternenseemann committed:
haskellPackages.goldplate: make executable available for tests

sternenseemann committed:
haskellPackages.hspec-api: downgrade to match hspec

cdepillabout committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

cdepillabout committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1792418](https://hydra.nixos.org/eval/1792418) of nixpkgs commit [9b7cbce](https://github.com/NixOS/nixpkgs/commits/9b7cbce8b25abb3de1c47fa5a655fbe559e86410) as of 2023-03-22 23:44 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
haskellPackages.hs-swisstable-hashtables-class: only build on x86_64

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
haskellPackages: add comments to release-haskell.nix

cdepillabout committed:
haskellPackages: small formatting fix to versionedCompilerJobs in release-haskell.nix

cdepillabout committed:
haskellPackages: small refactoring and comments to versionedCompilerJobs in release-haskell.nix

cdepillabout committed:
haskellPackages: make sure versionedCompilerJobs doesn't try running on platforms ghc does not support

cdepillabout committed:
haskellPackages: fix indentiation in hydra-report.hs

cdepillabout committed:
haskellPackages: add types and some formatting to hydra-report.hs

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
haskellPackages: add a little documentation to hydra-report.hs

cdepillabout committed:
haskellPackages: slight refactoring of hydra-report.hs

cdepillabout committed:
haskellPackages: add newtype for JobName in hydra-report.hs

This commits changes the `job` field in `Build` to a newtype.  This is
mostly just to have a place to document exactly what a job name consists
of.

cdepillabout committed:
haskellPackages: add newtype for PkgName and PkgSet in hydra-report.hs

Add a newtype for a package name and a package set.  This is less for
correctness, and more just to make the code a little easier to read
through without having to keep in mind what each Text refers to.

github-actions[bot] committed:
Merge master into haskell-updates

vcunat commented:
`.ffmpeg-light` will [break](https://hydra.nixos.org/log/fss5nzjklc08iz7759ha8bffq2vlmc6j-ffmpeg-light-0.14.1.drv) after the default ffmpeg moves from 4 to 5 (with PR #221461).  I assume that at least the initial step would be to just pin it to `ffmpeg_4`, but I don't know how to do that correctly, so perhaps you can do it?  (e.g. here)

cdepillabout committed:
haskellPackages: in hydra-report.hs, split Linux and Darwin build failures

This commit changes hydra-report.hs to split up Linux and Darwin build
failures into two different sections.  Darwin failures are hidden by
default.

cdepillabout committed:
haskellPackages: remove error about outstanding jobs on aarch64-darwin in hydra-report.hs

TristanCacqueray committed:
haskell.packages.ghc96.memory: bump version

This change enables using the memory package.

TristanCacqueray committed:
haskell.packages.ghc96.parallel: bump version

cdepillabout committed:
Merge pull request #223042 from cdepillabout/hydra-report-hide-darwin

haskellPackages: hide darwin jobs in hydra-report.hs

cdepillabout committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1792584](https://hydra.nixos.org/eval/1792584) of nixpkgs commit [d7bd936](https://github.com/NixOS/nixpkgs/commits/d7bd9361dd318ea459c610247fa2960676db6a5a) as of 2023-03-26 23:18 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

cdepillabout committed:
haskellPackages.ffmpeg-light: force use of ffmpeg-4

cdepillabout commented:
@vcunat Thanks for the heads up.  Fixed in 88e877fe1c7.

cdepillabout committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
Merge pull request #222190 from sternenseemann/vector-tests

haskell.packages.*.vector: clean up and unify overrides, run tests

TristanCacqueray committed:
haskell.packages.ghc96.foundation: add build patch

sternenseemann committed:
haskellPackages.cabal-install.scope.vector: don't pull in doctest

doctest depends on GHC which triggers inconsistent dependencies
similarly to hspec* < 2.10, so we need to avoid this. This problem newly
cropped up since we enabled tests for vector in #222190.

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
writers: fix aarch64-darwin workaround if content is a string

When authoring #220966, I missed that the `//` would take effect solely
in the else branch, fixing hledger-check-fancyassertions, but not
tests.writers.bin.haskell.

cdepillabout committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1792734](https://hydra.nixos.org/eval/1792734) of nixpkgs commit [192d03b](https://github.com/NixOS/nixpkgs/commits/192d03bee462f0eca2112c4f136fb1aac084f165) as of 2023-03-29 03:44 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

cdepillabout committed:
haskellPackages: filter out Darwin jobs from mergeable job in release-haskell.nix

cdepillabout committed:
haskellPackages.guardian: mark broken

It appears that a transitive dependency uses doctest, which is not
compatible with the Cabal library in scope for guardian.

cdepillabout committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates
